sale_product_country_restriction/models/sale_order_line.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+
+
+class SaleOrderLine(models.Model):
+
+    _inherit = "sale.order.line"
+
+    @api.onchange("product_id")
+    def _onchange_product_country_restriction(self):
+        """
+        Check if there are country restrictions on the product and get warning
+        message
+        :return: {'warning':...}
+        """
+        res = {}
+        if not self.env.company.enable_sale_country_restriction:
```

Comment:
Same here (almost correct, you should first use the company of the SO and if not set, the `self.env.company` is correct)